homeassistant/components/iota.py

```diff
@@ -0,0 +1,97 @@
+"""
+Support for IOTA wallets.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/iota/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import load_platform
+from homeassistant.helpers.entity import Entity
+
+DOMAIN = 'iota'
+
+REQUIREMENTS = ['pyota==2.0.3']
+
+IOTA_PLATFORMS = ['sensor']
+
+SCAN_INTERVAL = timedelta(minutes=10)
+
+CONF_IRI = 'iri'
+CONF_TESTNET = 'testnet'
+CONF_WALLETS = 'wallets'
+CONF_WALLET_NAME = 'name'
+CONF_WALLET_SEED = 'seed'
+
+WALLET_CONFIG = vol.Schema({
+    vol.Required(CONF_WALLET_NAME): cv.string,
+    vol.Required(CONF_WALLET_SEED): cv.string,
+})
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_IRI): vol.All(str),
+        vol.Optional(CONF_TESTNET, default=False): cv.boolean,
+        vol.Required(CONF_WALLETS): vol.All(cv.ensure_list, [WALLET_CONFIG])
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup(hass, config):
+    """Setup IOTA component."""
+    # Set domain sepecific data
+    iota_config = config[DOMAIN]
+    hass.data[DOMAIN] = {
+        'iri': iota_config[CONF_IRI],
+        'is_testnet': iota_config[CONF_TESTNET],
+        'wallets': iota_config[CONF_WALLETS]
+    }
+
+    # Set states of IRI config
+    hass.states.set('iota.iri', hass.data[DOMAIN]['iri'])
```

Comment:
Why do you want to set these states?

Reply:
This is the URL of the IOTA node to connect to, which may change. It is a separate state in order to see current URL of the node in the UI.

Reply:
Report it as a state attribute on the node sensor entity instead.

Reply:
done